<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Parameters of balloon detection -->
  <arg name="darknet_debug" default="false" />
  <arg name="balloon_frame" default="/map" />
  <arg name="n_points" default="20" />
  <arg name="lookup_limits" default="10" />
  <arg name="in_box_scale" default="0.8" />
  <arg name="search_radius" default="0.5" />
  <arg name="min_neighbors" default="10" />




  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>
  <arg name="image" default="/camera/rgb/image_raw" />

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file"             default="$(find darknet_ros)/config/ros.yaml"/>
  <arg name="network_param_file"         default="$(find darknet_ros)/config/balloon.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="darknet_ros" type="yolo_balloon_detector_node" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <remap from="camera/rgb/image_raw"  to="$(arg image)" />

    <param name="darknet_debug" value="$(arg darknet_debug)" />
    <param name="n_points" value="$(arg n_points)" />
    <param name="lookup_limits" value="$(arg lookup_limits)" />
    <param name="in_box_scale" value="$(arg in_box_scale)" />
    <param name="search_radius" value="$(arg search_radius)" />
    <param name="min_neighbors" value="$(arg min_neighbors)" />

  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
